src/visitors/infix_printer_visitor.py

```diff
@@ -108,3 +84,26 @@ def visit_real_variable(self, operator: RealVariable):
 
     def visit_real_equality(self, operator: RealEquality):
         return f"({operator.operator_1.accept(self)} = {operator.operator_2.accept(self)})"
+
+    def visit_string_concatenation(self, operator: StringConcatenation):
+        return f"({operator.operator_1.accept(self)} || {operator.operator_2.accept(self)})"
+
+    def visit_string_length(self, operator: StringLength):
+        return f"len({operator.operator_1.accept(self)})"
+
+    def visit_substring(self, operator: Substring):
+        return f"{operator.operator_1.accept(self)}" \
```

Comment:
I think here the name of the operation is missing

Reply:
(str.substr  s i j) is s[i, j] in the infix printer. Maybe you have a better idea.

Reply:
Sorry just miss read